nexus/auth/src/authn/external/session_cookie.rs

```diff
@@ -13,13 +13,15 @@ use chrono::{DateTime, Duration, Utc};
 use dropshot::HttpError;
 use http::HeaderValue;
 use nexus_types::authn::cookies::parse_cookies;
+use omicron_uuid_kinds::{ConsoleSessionKind, TypedUuid};
 use slog::debug;
 use uuid::Uuid;
 
 // many parts of the implementation will reference this OWASP guide
 // https://cheatsheetseries.owasp.org/cheatsheets/Session_Management_Cheat_Sheet.html
 
 pub trait Session {
+    fn id(&self) -> TypedUuid<ConsoleSessionKind>;
```

Comment:
if you took my advice above, this would be
```suggestion
    fn id(&self) -> ConsoleSessionUuid;
```